nixos/modules/system/boot/networkd.nix

```diff
@@ -2077,6 +2077,26 @@ let
       ];
       kernelModules = [ "af_packet" ];
 
+      systemd.services.nixos-flush-networkd = mkIf config.boot.initrd.network.flushBeforeStage2 {
+        description = "Flush Network Configuration";
+        wantedBy = ["initrd.target"];
+        after = ["systemd-networkd.service" "dbus.socket" "dbus.service"];
```

Comment:
It's *possible* we should have an ordering between this service and `initrd-switch-root.target`. When there's any ordering at all between two units, and a transaction requires one to be stopped and one to be started (such as `systemctl isolate initrd-switch-root.target`), the stopping units are always stopped before the starting units are started. But that's only true if there's an ordering between them. Otherwise the stopping and starting may be simultaneous, which *I think* is bad in this case?

Reply:
@oddlama whether or not the unit is stopped isn't the issue. The issue is whether it's stopped *before* `initrd-switch-root.target` is reached. When systemd isolates a unit, it *does* take the stop action on all units that aren't wanted by the unit being isolated, but that doesn't mean they'll be stopped *first*. They may end up being stopped after the unit being isolated has been started.

Reply:
I see. So i dug a little deeper and it seems you are absolutely right, this would execute in parallel if no explicit dependencies are defined. There's a systemd issue about a related topic which [talks about this a bit in (and following comment)](https://github.com/systemd/systemd/issues/3436#issuecomment-224503276). Not sure whether we should also think about the `DefaultDependencies=no` stuff mentioned there, but from what I read adding an order would definitely be a good idea.